src/java.base/share/classes/java/io/Reader.java

```diff
@@ -391,6 +391,41 @@ public int read(char[] cbuf) throws IOException {
      */
     public abstract int read(char[] cbuf, int off, int len) throws IOException;
 
+
+    /**
+     * Reads all remaining characters into a string.
+     *
+     * <p> This method reads all content including the line separators in
+     * the middle and/or at the end. The resulting string will contain line
+     * separators as they appear in the original content. The method does not
+     * close this reader nor its underlying stream. If an I/O error occurs,
+     * the states of the reader and its underlying stream are unspecified.
+     *
+     * @apiNote
+     * This method is intended for simple cases where it is appropriate and
+     * convenient to read all lines into a String. It is not intended for
+     * reading a large number of lines.
+     *
+     * @return     a String containing all lines
+     *
+     * @throws     IOException       If an I/O error occurs
+     * @throws     OutOfMemoryError  If the content is extremely large,
+     *                               for example larger than {@code 2GB}
+     *
+     * @see java.nio.file.Files#readString
+     *
+     * @since 25
+     */
+    public String readString() throws IOException {
+        StringBuilder result = new StringBuilder();
+        char[] str = new char[TRANSFER_BUFFER_SIZE];
+        int n;
+        while ((n = read(str)) != -1) {
+            result.append(str, 0, n);
```

Comment:
If the characters in the Reader are always Latin1, this process will convert the UTF16 char[] to the LATIN1 encoded byte[] value in the StringBuilder every time, which will cause unnecessary performance overhead.

Reply:
We need to provide a StringBuilder that creates a coder of UTF16 to avoid the unnecessary UTF16 to LATIN1 conversion overhead here. 

If we do not want to add a new public method, we can add an internal method and expose it through JLA for use here.

Reply:
`Reader` by specification works on `char`, so I think this is fine. Even if a StringBuilder is UTF16, it still has to check for all-Latin1 characters at the time of string creation, so not really an overhead elimination.

Reply:
Seems you want to make the `result` StringBuilder fully expanded (maybeLatin1 = true, coder = UTF16) initially, so we just compress the whole builder with a vectorized intrinsic instead of char-by-char loop in `append`. Makes sense.

Reply:
```java
    public static String readString() throws IOException {
        char[] chars = new char[TRANSFER_BUFFER_SIZE];
        int n;
        int off = 0;
        int rest = chars.length;
        while ((n = read(chars, off, rest)) != -1) {
            off += n;
            if (n == rest) {
                chars = Arrays.copyOf(chars, chars.length * 2);
            }
            rest = chars.length - off;
        }
        return new String(chars, 0, off);
    }
```
Maybe this version is better, it directly expands the capacity on char[] without using StringBuilder, has good performance, and does not need to add new methods.

Reply:
Maybe it can be implemented by referring to `InputStream::readNBytes(int)` (The default implementation of `InputStream::readAllBytes()` is based on it):

https://github.com/openjdk/jdk/blob/22e8a97a1ce4e1c781fbc6f1e271c477fe95f069/src/java.base/share/classes/java/io/InputStream.java#L396-L458

Reply:
https://github.com/openjdk/jdk/pull/24728#issuecomment-2815687444